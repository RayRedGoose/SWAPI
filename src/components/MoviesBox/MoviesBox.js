import React, { Component } from 'react';
import MovieCard from '../MovieCard/MovieCard';
import './MoviesBox.scss';

export default class MovieBox extends Component {
  constructor() {
    super()
    this.state = {
      cards: []
    }
  }

  getMovieInfo = () => {
    return null
  }
  generateCards = () => {
    return null
  }

  render() {
    return (
      <section className="movie-box">
        {this.generateCards()}
      </section>
    )
  }
}
