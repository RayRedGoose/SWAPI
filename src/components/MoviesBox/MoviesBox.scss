.movie-box {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  overflow: scroll;
  padding-bottom: 10em;
}
