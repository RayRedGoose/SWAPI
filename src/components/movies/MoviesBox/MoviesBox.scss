.movie-box {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  overflow: scroll;
  padding: 0 20px;
  margin: 0 20px;
}
