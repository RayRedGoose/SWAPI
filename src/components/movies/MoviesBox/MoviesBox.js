import React, { Component } from 'react'
import MovieCard from '../MovieCard/MovieCard'
import PropTypes from 'prop-types'
import './MoviesBox.scss'

export default class MovieBox extends Component {
  constructor(props) {
    super(props)
    this.state = {}
  }

  componentDidMount = () => {
    this.props.addMovies()
  }

  generateCards = () => {
    return this.props.movies.map((movie, index) => (
      <MovieCard
        key={index}
        id={index+1}
        {...movie} />
      )
    )
  }

  render() {
    return (
      <section className="movie-box">
        { this.generateCards() }
      </section>
    )
  }
}

MovieBox.propTypes = {
  movies: PropTypes.arrayOf(PropTypes.object).isRequired,
  addMovies: PropTypes.func.isRequired
}
