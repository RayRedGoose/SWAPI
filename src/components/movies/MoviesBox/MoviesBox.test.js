import React from 'react'
import { shallow } from 'enzyme'
import MoviesBox from './MoviesBox'

describe("MoviesBox", () => {
  const mockProps = [{
    title: "A Star War",
    episode_id: 4,
    director: "The Rock",
    producer: "Mr. Tumnus",
    release_date: "1977-05-25",
    opening_crawl: "paaaam-pam paaaaam-pam pa-pa-paaaam"
  }]

  const addMovies = jest.fn()

  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <MoviesBox
        movies={mockProps}
        addMovies={addMovies}
      />
    )
  })

  it("should render MoviesBox", () => {
    expect(wrapper).toMatchSnapshot()
  })

  it("should call addMovies props after rendering", () => {
    expect(addMovies).toHaveBeenCalled()
  })

  it("should call generateCards after rendering", () => {
    const spy = jest.spyOn(wrapper.instance(), 'generateCards').mockImplementation(arg => arg)
    wrapper.instance().forceUpdate()

    expect(spy).toHaveBeenCalled()
  })
})
