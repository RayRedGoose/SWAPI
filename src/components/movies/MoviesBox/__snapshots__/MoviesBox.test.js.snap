// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesBox should render MoviesBox 1`] = `
<section
  className="movie-box"
>
  <MovieCard
    director="The Rock"
    episode_id={4}
    id={1}
    key="0"
    opening_crawl="paaaam-pam paaaaam-pam pa-pa-paaaam"
    producer="Mr. Tumnus"
    release_date="1977-05-25"
    title="A Star War"
  />
</section>
`;
